src/dotnet-roslyn-tools/Commands/DartTestCommand.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the License.txt file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.RoslynTools.Utilities;
+using Microsoft.RoslynTools.VS;
+
+namespace Microsoft.RoslynTools.Commands;
+
+internal static class DartTestCommand
+{
+    private static readonly string[] s_allProductNames = VSBranchInfo.AllProducts.Select(p => p.Name.ToLower()).Concat(new[] { "all" }).ToArray();
+
+    private static readonly DartTestCommandDefaultHandler s_dartTestCommandHandler = new();
+    private static readonly Option<int> prNumber = new(["--prNumber", "-pr"], "PR number") { IsRequired = true };
+    private static readonly Option<string> sha = new(["--sha", "-s"], "Relevant SHA") { IsRequired = false };
+    private static readonly Option<string> productOption = new Option<string>(new[] { "--product", "-p" }, () => "roslyn", "Which product to get info for").FromAmong(s_allProductNames);
+
+    public static Symbol GetCommand()
+    {
+        var command = new Command("dart-test",
+            @"Runs the dartlab pipeline for a given PR number and SHA.
+It works by cloning the PR from the dotnet/roslyn repo into the internal mirror and then running the dartlab pipeline from it.")
+        {
+            prNumber,
+            sha,
+            productOption,
+            CommonOptions.GitHubTokenOption,
+            CommonOptions.DevDivAzDOTokenOption,
+            CommonOptions.DncEngAzDOTokenOption,
+        };
+
+        command.Handler = s_dartTestCommandHandler;
+        return command;
+    }
+
+    private class DartTestCommandDefaultHandler : ICommandHandler
+    {
+        public async Task<int> InvokeAsync(InvocationContext context)
+        {
+            var logger = context.SetupLogging();
+            var settings = context.ParseResult.LoadSettings(logger);
+
+            if (string.IsNullOrEmpty(settings.GitHubToken) /*||
+                string.IsNullOrEmpty(settings.DevDivAzureDevOpsToken) ||
+                string.IsNullOrEmpty(settings.DncEngAzureDevOpsToken)*/)
```

Comment:
Reformat as necessary. We can have empty ADO tokens if we are running in interactive authentication mode.

```suggestion
            if (string.IsNullOrEmpty(settings.GitHubToken) ||
                (settings.IsCI && 
                    (string.IsNullOrEmpty(settings.DevDivAzureDevOpsToken) ||
                     string.IsNullOrEmpty(settings.DncEngAzureDevOpsToken))))
```